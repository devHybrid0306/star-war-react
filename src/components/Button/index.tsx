import React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  width: auto;
  height: 50px;
  background-color: ${({ theme }) => theme.color.copper_coin};
  color: ${({ theme }) => theme.color.white};
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid ${({ theme }) => theme.color.copper_coin};
  border-radius: 3px;
`;

interface IButtonProps {
  text: string;
}

const Button: React.FC<IButtonProps> = ({ text }) => {
  return <StyledButton>{text}</StyledButton>;
};

export default Button;
