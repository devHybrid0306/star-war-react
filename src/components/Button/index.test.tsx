import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';

import Button from '.';

test('testing Button', () => {
  const { container } = render(<Button text="button" />);

  const button = screen.getByText('button');
  expect(button).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
