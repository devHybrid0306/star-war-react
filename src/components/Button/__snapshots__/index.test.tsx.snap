// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button 1`] = `
.c0 {
  width: auto;
  height: 50px;
  background-color: 'rgb(219, 54, 21)';
  color: white;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid 'rgb(219, 54, 21)';
  border-radius: 3px;
}

<div>
  <button
    class="c0"
  >
    button
  </button>
</div>
`;
